src/Core/Localization/Specification/Factory.php

```diff
@@ -113,7 +113,7 @@ public function buildPriceSpecification(
             $this->getNegativePattern($currencyPattern),
             $this->computeNumberSymbolLists($numbersSymbols),
             $maxFractionDigits ?? $this->getMaxFractionDigits($positivePattern),
-            $this->getMinFractionDigits($positivePattern),
+            is_int((int)$currency->getDecimalPrecision()) ? (int)$currency->getDecimalPrecision(): $this->getMinFractionDigits($positivePattern),
```

Comment:
```suggestion
            is_int($currency->getDecimalPrecision()) ? (int) $currency->getDecimalPrecision(): $this->getMinFractionDigits($positivePattern),
```

Reply:
this is the first version of the code and does not work, this will result the screenshots that @khouloudbelguith has sent above. We need this `int` cast to run `(int)$currency->getDecimalPrecision()` and fix the issue.
Thanks

Reply:
So the check is useless as casting something in int will always return an int oO

Reply:
In the previous [PR](https://github.com/PrestaShop/PrestaShop/pull/23912) @sowbiba changed this line.

Reply:
I think the check has to be

is_int(toto) ? toto : somethingElse
No need to cast toto

Doing this, precision === '4' will result with *somethingElse*

If you want to allow strings that "look like integer" you can do
ctype_digit(strval(toto)) ? (int) toto : somethingElse

Reply:
```
var_dump(is_int(42)); //true
var_dump(is_int('42')); //false
var_dump(is_int('42.5')); //false
var_dump(is_int('42,5')); //false

var_dump(is_numeric(42)); //true
var_dump(is_numeric('42')); //true
var_dump(is_numeric('42.5')); //true
var_dump(is_numeric('42,5')); //false

var_dump(ctype_digit(42)); //false
var_dump(ctype_digit('42')); //true
var_dump(ctype_digit('42.5')); //false
var_dump(ctype_digit('42,5')); //false

Reply:
@sowbiba thanks, I will check the other forms to check the int val..